samples/dml.js

```diff
@@ -0,0 +1,666 @@
+      })
+      .then(() => {
+        // Close the database when finished.
+        return database.close();
+      });
+  });
+
+  // [END spanner_dml_getting_started_insert]
+}
+
+function writeWithTransactionUsingDml(instanceId, databaseId, projectId) {
+  // [START spanner_dml_getting_started_update]
+  // This sample transfers 200,000 from the MarketingBudget field
+  // of the first Album to the second Album. Make sure to run the
+  // addColumn and updateData samples first (in that order).
+
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  const transferAmount = 200000;
+  const minimumAmountToTransfer = 300000;
+
+  database.runTransaction((err, transaction) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    let firstBudget, secondBudget;
+    const queryOne = `SELECT MarketingBudget FROM Albums
+      WHERE SingerId = 1 AND AlbumId = 1`;
+
+    const queryTwo = `SELECT MarketingBudget FROM Albums
+    WHERE SingerId = 2 AND AlbumId = 2`;
+
+    Promise.all([
+      // Reads the first album's budget
+      transaction.run(queryOne).then(results => {
+        // Gets first album's budget
+        const rows = results[0].map(row => row.toJSON());
+        firstBudget = rows[0].MarketingBudget;
+        console.log(`The first album's marketing budget: ${firstBudget}`);
+
+        // Makes sure the first album's budget is sufficient
+        if (firstBudget < minimumAmountToTransfer) {
+          throw new Error(
+            `The first album's budget (${firstBudget}) is less than the minimum required amount to transfer.`
+          );
+        }
+      }),
+
+      // Reads the second album's budget
+      transaction.run(queryTwo).then(results => {
+        // Gets second album's budget
+        const rows = results[0].map(row => row.toJSON());
+        secondBudget = rows[0].MarketingBudget;
+        console.log(`The second album's marketing budget: ${secondBudget}`);
+      }),
+    ])
+      .then(() => {
+        // Transfer the budgets between the albums
+        console.log(firstBudget, secondBudget);
+        secondBudget += transferAmount;
+        firstBudget -= transferAmount;
+
+        console.log(firstBudget, secondBudget);
+
+        // Update the database
+        // Note: Cloud Spanner interprets Node.js numbers as FLOAT64s, so they
+        // must be converted (back) to strings before being inserted as INT64s.
+
+        transaction.runUpdate({
```

Comment:
Do we want a `return` or `Promise.all()` for these two calls?

Reply:
Leaving as is for now as this follows the pattern in transactions.js

Reply:
I might be reading it wrong, but it appears to run two concurrent updates and the promise chain does not wait for either to finish before committing the transaction. The closest I could find in transaction.js to this was the [usage of `update`](https://github.com/googleapis/nodejs-spanner/blob/master/samples/transaction.js#L176) which is a synchronous method whereas `runUpdate` is async.